synapse/handlers/federation.py

```diff
@@ -923,7 +923,8 @@ async def backfill(self, dest, room_id, limit, extremities):
                 )
             )
 
-        await self._handle_new_events(dest, ev_infos, backfilled=True)
+        if ev_infos:
```

Comment:
Is this if-statement just an optimization?

Reply:
Oh err, somewhere along the lines I think something got very unhappy about empty lists. I forget the details now or if its even necessary :confused: